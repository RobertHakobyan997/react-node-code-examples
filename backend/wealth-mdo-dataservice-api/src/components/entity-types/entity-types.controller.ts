import { Controller, Get } from '@nestjs/common';
import { ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
import { Log } from 'ngpd-merceros-wealth-mdo-common-be/dist/logger/mdo-logger';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';

import { EntityTypesService } from './entity-types.service';

@ApiTags('Entity Types')
@Controller('entity-types')
export class EntityTypesController {
  constructor(private readonly service: EntityTypesService) {
  }

  @ApiOperation({
    description: 'Get all entity types',
  })
  @ApiOkResponse({
    description: 'Returns all entity types',
    type       : [ Enum ],
  })
  @Get()
  @Log('Get supported entity types')
  findAll() {
    return this.service.findAll();
  }
}
