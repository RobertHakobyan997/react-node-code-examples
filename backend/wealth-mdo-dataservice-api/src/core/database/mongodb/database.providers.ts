import mongoose from 'mongoose';
import { Logger } from '@nestjs/common';
import { IS_LOCAL } from 'ngpd-merceros-wealth-mdo-common-be/dist/utils/environments';

import mongoConfig from '../../../config/mongo.config';

export const databaseProviders = [
  {
    // ToDo: change to dynamic provider and take it from mongoConfig
    provide   : 'DatabaseConnection',
    useFactory: (): Promise<typeof mongoose> => {
      if (IS_LOCAL()) {
        mongoose.set('debug', (collectionName, methodName, ...methodArgs) => {
          const logger = new Logger(collectionName);
          const args = methodArgs.reduce((acc, next, idx) => {
            acc += JSON.stringify(next, null, 2) + (idx === methodArgs.length - 1 ? '' : ', ');
            return acc;
          }, '');
          logger.log(`${methodName}(${args})`);
        });
      }
      return mongoose.connect(mongoConfig().uri, mongoConfig().options);
    },
  },
];
