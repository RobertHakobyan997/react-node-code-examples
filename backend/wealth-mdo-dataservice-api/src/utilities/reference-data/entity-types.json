[
  {
    "_id": "ObjectId('608ba9cbc1656e7b68ab3ed3')",
    "key": "fund",
    "display": "Fund",
    "order": 1
  },
  {
    "_id": "ObjectId('608ba9cbc1656e7b68ab3ed4')",
    "key": "client",
    "display": "Client",
    "order": 2
  },
  {
    "_id" : "ObjectId('611a61778699cfd5d32d81b7')",
    "key" : "account",
    "display" : "Account",
    "order" : 3
  },
  {
    "_id" : "ObjectId('611a61778699cfd5d32d81b2')",
    "key": "portfolioBB",
    "display": "Portfolios & BB",
    "order" : 4
  },
  {
    "_id" : "ObjectId('6184dc46ef402f4bab802361')",
    "key": "subAdvisor",
    "display": "Sub Advisor",
    "order" : 5
  },
  {
    "_id": "ObjectId('6184dc46ea402f4bab802362')",
    "key": "securityHoldingInformation",
    "display": "Security Holding Information",
    "order": 6
  },
  {
    "_id": "ObjectId('6184dc46ea402f4bab802372')",
    "key": "subIm",
    "display": "Sub IM",
    "order": 7
  },
  {
    "_id": "ObjectId('62138814dd3e7e07833fc9fd')",
    "key": "referenceData",
    "display": "Reference Data",
    "order": 8
  }
]
