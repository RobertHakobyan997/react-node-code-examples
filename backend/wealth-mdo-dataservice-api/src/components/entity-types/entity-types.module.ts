import { Module } from '@nestjs/common';
import EntityTypesModel, { EntityTypesModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/entity-types';

import { DatabaseModule } from '../../core/database/mongodb/database.module';

import { EntityTypesController } from './entity-types.controller';
import { EntityTypesService } from './entity-types.service';
import { EntityTypesDal } from './entity-types.dal';

@Module({
  imports    : [ DatabaseModule ],
  controllers: [ EntityTypesController ],
  providers  : [
    { provide: EntityTypesModelToken, useFactory: () => EntityTypesModel },
    EntityTypesService,
    EntityTypesDal
  ],
  exports: [ EntityTypesService ],
})
export class EntityTypesModule {
}
