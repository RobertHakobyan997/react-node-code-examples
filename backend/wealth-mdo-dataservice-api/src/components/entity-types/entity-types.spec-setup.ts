import { Test, TestingModule } from '@nestjs/testing';
import { EntityTypesModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/entity-types';

import { mockMongooseModel } from '../../../test/mocks/mock-mongoose.model';

import { EntityTypesService } from './entity-types.service';
import { EntityTypesDal } from './entity-types.dal';

export const assert = async () => {
  const module: TestingModule = await Test.createTestingModule({
    providers: [
      EntityTypesService,
      EntityTypesDal,
      { provide: EntityTypesModelToken, useValue: mockMongooseModel },
    ],
  }).compile();

  return module.get<EntityTypesService>(EntityTypesService);
};
