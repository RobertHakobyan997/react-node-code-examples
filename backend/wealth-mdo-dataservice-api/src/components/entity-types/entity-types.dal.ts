import { Inject, Injectable } from '@nestjs/common';
import { FilterQuery, Model, UpdateQuery } from 'mongoose';
import { Logger } from 'ngpd-merceros-logger-be';
import { plainToClass } from 'class-transformer';
import EntityTypesModel, { EntityTypesModelToken } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/entity-types';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';

@Injectable()
export class EntityTypesDal {
  private readonly logger = new Logger(EntityTypesDal.name);

  constructor(@Inject(EntityTypesModelToken) private readonly model: Model<Document<Enum>>) {
  }

  findAll(filter: FilterQuery<Document<Enum>> = {}, sort: string | any = {}, options: any = {}) {
    return this.model
      .find(filter, {}, options)
      .sort(sort)
      .exec();
  }

  findOne(filter: FilterQuery<Document<Enum>> = {}) {
    return this.model.findOne(filter).exec();
  }

  findById(id: any, options: any = {}) {
    return this.model.findById(id, {}, options).exec();
  }

  insert(docs: any[]) {
    return this.model.insertMany(plainToClass(EntityTypesModel, docs));
  }

  findByIdAndUpdate(id: any, docs: UpdateQuery<Enum>) {
    return this.model.findByIdAndUpdate(id, plainToClass(EntityTypesModel, docs));
  }

  findByIdAndDelete(id: any) {
    return this.model.findByIdAndDelete(id);
  }
}
