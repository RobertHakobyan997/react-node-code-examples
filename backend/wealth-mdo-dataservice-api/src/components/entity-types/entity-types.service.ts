import { Injectable } from '@nestjs/common';
import { FilterQuery } from 'mongoose';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';

import { EntityTypesDal } from './entity-types.dal';

@Injectable()
export class EntityTypesService {
  constructor(private readonly dal: EntityTypesDal) {
  }

  findAll(filter: FilterQuery<Document<Enum>> = {}, sort: string | any = {}, options: any = {}) {
    return this.dal.findAll(filter, sort, options);
  }
}
