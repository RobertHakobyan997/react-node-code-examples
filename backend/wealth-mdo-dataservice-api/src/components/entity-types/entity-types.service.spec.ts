import { Document } from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/mongodb/document';
import Enum from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/structures/enum';
import { plainToClass } from 'class-transformer';
import EntityTypesModel from 'ngpd-merceros-wealth-mdo-common-be/dist/domain/model/mongoose/entity-types';
import { Types } from 'mongoose';

import { entityTypes, entityTypesRaw } from '../../../test/data/entity-types/entity-types.const';

import { EntityTypesService } from './entity-types.service';
import { assert } from './entity-types.spec-setup';

describe('EntityTypesService', () => {
  let service: EntityTypesService;

  beforeEach(async () => {
    service = await assert();
  });

  it('plainToClass', () => {
    const mapped = plainToClass(EntityTypesModel, entityTypesRaw);
    expect(mapped._id).toBeInstanceOf(Types.ObjectId);
    expect(mapped).toMatchSnapshot();
  });

  describe('getEntityTypes()', () => {
    it('should return all entity types', async () => {
      jest.spyOn(service['dal'], 'findAll').mockResolvedValue(entityTypes as Document<Enum>[]);
      const res = await service.findAll();
      expect(res).toEqual(entityTypes);
      expect(res.length).toBe(1);
    });
  });
});
