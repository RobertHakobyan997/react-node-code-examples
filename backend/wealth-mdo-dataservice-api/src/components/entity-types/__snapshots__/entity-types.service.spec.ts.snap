// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityTypesService plainToClass 1`] = `
Object {
  "_id": "609d4c33bfd3901b8c3af7f7",
  "display": "Fund",
  "hidden": false,
  "key": "fund",
  "order": 1,
}
`;
